src/base/settingvalue.h

```diff
@@ -44,25 +44,25 @@ class CachedSettingValue
     using ProxyFunc = std::function<T (const T&)>;
 
 public:
-    explicit CachedSettingValue(const char *keyName, const T &defaultValue = T())
+    explicit constexpr CachedSettingValue(const char *keyName, const T &defaultValue = T())
```

Comment:
I confess I do not understand `constexpr`. But can we use this at compile-time if it depends on an object that is not available?
IMO, `constexpr` is unusable with this class.

Reply:
The `constexpr` lets the compiler **try** to init it at compile-time, it doesn't matter if it could fail doing so at compile-time.

>IMO, constexpr is unusable with this class.

Seems yes in current state (due to QString), however an (wild?) idea is turn m_keyName to QLatin1String type, this way it is possible to initialize the class complie-time, what do you think?

Reply:
>however an (wild?) idea is turn m_keyName to QLatin1String type, this way it is possible to initialize the class complie-time, what do you think?

I don't think that will help. It's not about QString.
The problem is that to initialize it, we have to use a QSettingsStorage instance that is initialized at run time, and it cannot be initialized at compile time, because it has to read the file from disk.

Reply:
OK, I removed that part.